libnymea/types/actiontype.h

```diff
@@ -49,6 +49,9 @@ class LIBNYMEA_EXPORT ActionType
     ParamTypes paramTypes() const;
     void setParamTypes(const ParamTypes &paramTypes);
 
+    static QStringList jsonProperties();
```

Comment:
I must say I don't like this too much... I get the idea behind it and it makes sense t keep things in sync between the actual containers and their json. However, this requires those data containers to be aware of json while that shouldn't be the case. Imagine we want to add another way of writing plugins, say, python plugins and for some reason that uses yaml instead of json, this kinda breaks.

Idea: could we make those classes Q_GADGETs and use the Q_PROPERTY macro instead to mark such properties, and then use the QMetaObject's properties to check required fields in the json/yaml/whatever?

That would also match the long therm plan of eventually being able to provide libnymea to write client applications. Since Qt 5.10 QML is able to use Q_GADGETS too, not only QObjects, so that would make sense. 

Not sure right now how we could mark optional vs mandatory properties with that, but maybe we can discuss this thought further?

wdyt?